lxd/certificates.go

```diff
@@ -1039,3 +1034,26 @@ func certificateDelete(d *Daemon, r *http.Request) response.Response {
 
 	return response.EmptySyncResponse
 }
+
+func certificateValidate(cert *x509.Certificate) error {
+	if time.Now().Before(cert.NotBefore) {
+		return fmt.Errorf("The provided certificate isn't valid yet")
+	}
+
+	if time.Now().After(cert.NotAfter) {
+		return fmt.Errorf("The provided certificate is expired")
+	}
+
+	if cert.PublicKeyAlgorithm == x509.ECDSA {
+		pubKey, ok := cert.PublicKey.(*rsa.PublicKey)
```

Comment:
Maybe I'm wrong, but `x509.ECDSA` shouldn't have any `rsa.Publickey`, no?

Reply:
yeah, was meant to be == x509.RSA